### ISSUE TITLE
Convert-TestResultsToJUnit.ps1 should not treat test failures as an error

### ISSUE THREAD

**Issue by danieljurek** [Azure/azure-sdk-for-rust#3964]
The `cargo2junit` tool currently exits with a nonzero exit code if there is a test failure. Since CI runs tests and JUnit conversion in separate tasks, this failure signal is redundant and can be ignored. The script should only fail if conversion itself fails. 

The current implementation of cargo2junit fails if a test failure is detected: 

```rust
fn determine_exit_code(report: &Report) -> Result<()> {
    if report.testsuites().is_empty() {
        Err(Error::new(ErrorKind::NotFound, "No test suite results were found.".to_owned()))
    } else if report
        .testsuites()
        .iter()
        .flat_map(|suite| suite.testcases().iter())
        .any(|testcase| testcase.is_error() || testcase.is_failure())
    {
        Err(Error::other("One or more tests failed.".to_owned()))
    } else {
        Ok(())
    }
}
```

stderr in the logs includes the following: 

```
Error: Custom { kind: Other, error: "One or more tests failed." }
```

To prevent noise, use a different algorithm: 

* If cargo2junit exits 0: continue
* If cargo2junit exits non-zero: 
    * If stderr only has the "One or more tests failed." error: continue (do not treat as a failure) 
    * If stderr contains any additional errors: treat as a failure and set `$succeeded = $false` 

Don't do any clever parsing of stderr, it's enough to assume that there's one error per-line. Make sure that stderr logs are still printed so that the CI logs contain whatever came out of stderr.. I want to be able to debug in the event of a future break.
